client/core/bond.go

```diff
@@ -499,7 +499,7 @@ func (c *Core) preValidateBond(dc *dexConnection, bond *asset.Bond) error {
 		return fmt.Errorf("account keys not decrypted")
 	}
 
-	assetID, bondCoin := bond.AssetID, bond.CoinID
+	assetID, bondCoin := bond.AssetID, bond.UnsignedCoinID
```

Comment:
Since we only store a `db.Bond` after `preValidateBond` has succeeded, I suspect we don't need to modify any of the DB code to store this.

Looking at how this coinID is used, does it seem kinda worthless?  Is the only place it's used just below with `if !bytes.Equal(preBondRes.BondID, bondCoin) {`?

Reply:
Say we remove `CoinID` from `msgjson.PreValidateBondResult`, it somewhat devalues the message's signature as it doesn't necessarily indicate they've validated a specific transaction, but we can have the signature be an amalgamation of the request and the response fields.  e.g.

```go
	preBondRes := &msgjson.PreValidateBondResult{
		AccountID: acctID[:],
		AssetID:   assetID,
		Amount:    uint64(amt),
		Expiry:    uint64(expireTime.Unix()),
	}
	preBondRes.SetSig(auth.SignMsg(append(preBondRes.Serialize(), preBond.RawTx...)))
```

In some places we actually have the signature pertain to the request, which is admittedly generally expected, but it could work I think.

Client would obviously have to change their interpretation of `msgjson.PreValidateBondResult.Signature` so that could be a pain, but at least we aren't released yet.

Reply:
How about instead of returning the `BondID` in the `PreValidateBondResult`, it just returns a `RawTransactionHash` which is always a sha256 hash of the raw transaction sent in the request.

Reply:
I gave this a thought too, but it's basically what `SignMsg` is doing, in that it does a `sha256` hash of some data.  If this data includes `RawTx`, you're validating that by validating the signature.  No need to actually send the sha256 of RawTx

Reply:
Yeah they both work the same way, I just thought it would be good to keep it consistent that the signature in every response is just signing the response object.

Reply:
So: (1) don't return `BondID` in the `PreValidateBondResult`, (2) don't return `UnsignedCoinID` in `asset.Bond`, and (3) don't store it in `db.Bond` and the DB.

Reply:
> Yeah they both work, I just thought it would be good to keep it consistent that the signature in every response is just signing the response object.

Understood.  We're already not consistent on this though. (See connect response signature)
We could be more conventional however.  I just thing this coin ID is essentially meaningless, and it's quite a burden to pass around.  
I wouldn't be totally opposed to returning a `RawTxHash` in the manner you've described though.
Mainly I want to not have to deal with `bond.UnsignedCoinID` at any level.